src/main/datatypes/MQ2BasicTypes.cpp

```diff
@@ -1672,6 +1674,13 @@ bool MQ2TimeType::GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQ
 		Dest.Type = pIntType;
 		return true;
 
+	case TimeMembers::MilliSecond:
+		SYSTEMTIME nowMs;
+		GetSystemTime(&nowMs);
+		Dest.DWord = nowMs.wMilliseconds;
+		Dest.Type = pIntType;
+		return true;
+
```

Comment:
This isn't quite right because you're not using the backing data (in this case VarPtr.Ptr) to get this value. I understand that the tm struct doesn't have ms available, but to do this correctly, you need to change the backing data from tm to something that does have ms, and then update all the members.

Right now this can be confusing because someone could declare a Time object expecting the ms to be from when they declared it, but this would give ms from the time they accessed it instead.

Reply:
I reworked internals to use SYSTEMTIME. I'm willing to change it to chrono, but I am not sure where to start.

Before:


![time-before](https://user-images.githubusercontent.com/181531/194164078-854a7ac7-c85c-4353-b12b-b24d711d234b.jpg)

After:

![time-after](https://user-images.githubusercontent.com/181531/194164093-a67a93d6-f081-4362-8b72-570e9ebbaf64.JPG)



```lua

local mq = require("mq")

print("Time: ", mq.TLO.Time())
print("Time.Hour: ", mq.TLO.Time.Hour())
print("Time.Minute: ", mq.TLO.Time.Minute())
print("Time.Second: ", mq.TLO.Time.Second())
--print("Time.MilliSecond: ", mq.TLO.Time.MilliSecond())
print("Time.DayOfWeek: ", mq.TLO.Time.DayOfWeek())
print("Time.Day: ", mq.TLO.Time.Day())
print("Time.Month: ", mq.TLO.Time.Month())
print("Time.Year: ", mq.TLO.Time.Year())
print("Time.Time12: ", mq.TLO.Time.Time12())
print("Time.Time24:", mq.TLO.Time.Time24())
print("Time.Date: ", mq.TLO.Time.Date())
print("Time.Night: ", mq.TLO.Time.Night())
print("Time.SecondsSinceMidnight: ", mq.TLO.Time.SecondsSinceMidnight())
print("Time.Hour12: ", mq.TLO.Time.Hour12())

```